### PR TITLE
#2661 Retry SMS ThrottlingException in retry-tasks queue

### PR DESCRIPTION
**Pull request by ChrisJohnson-CDJ**
# Description

issue #2661 

Explicitly retry SMS delivery ClientError(ThrottlingException) in the non-priority retry-queue.
This helps alleviate situations where a large number of SMS requests are getting throttled by AWS.
AWS Botocore will auto-retry but relies on a blocking sleep, holding the worker thread.
Explicitly retrying (once Botocore returns the exception) in the retry-queue will prevent the priority worker pool from being consumed by "sleeping" worker tasks.

The RetryableException was updated to add a `use_non_priority_handling` flag to act as a hint/recommendation for handlers.

- EUM (PinpointV2) ClientError(ThrottlingException) is raised as a RetryableException with a hint to retry as non-priority
    - `raise RetryableException(use_non_priority_handling=True)`
- Exception flows back up to deliver_sms or deliver_sms_with_rate_limiting and the V2 error handler calls task.retry instead of raising AutoRetry
- Explicit 'demotion' to `retry-tasks` queue logged in retry warning message prior to calling celery_task.retry
    - `current_app.logger.warning('%s unable to send for notification %s, retrying in non-priority worker pool',...)`
- The countdown is calculated the same way exponential backoff is handled in Celery
    - https://github.com/celery/celery/blob/87db973f08dba416a0a605ed7fd72e47f5dbd5bd/celery/utils/time.py#L448
    - https://github.com/celery/celery/blob/41bad6f22b66a4e510a58bf7130935cf0916f9db/celery/app/autoretry.py#L33

## How Has This Been Tested?

- Updated unit test to ensure the two retryable ClientErrors are handled properly (normal vs. non-priority)
- Added unit tests for high level provider_tasks exception handling to test manual (non-priority) retry vs. AutoRetry
- local tests run clean
- GHA tests run clean
- Dev deploy, regressions pass
- Sanity check sending SMS / Email (to be safe), delivered
    - SMS Delivered `host:*dev* ("*eb034704-fca5-437a-9637-757615212719*" OR @requestId:eb034704-fca5-437a-9637-757615212719 OR "*ff620735-8356-44de-9b62-a5a31547ea8d*") `
    - [Datadog SMS](https://vanotify.ddog-gov.com/logs?query=host%3A%2Adev%2A%20%28%22%2Aeb034704-fca5-437a-9637-757615212719%2A%22%20OR%20%40requestId%3Aeb034704-fca5-437a-9637-757615212719%20OR%20%22%2Aff620735-8356-44de-9b62-a5a31547ea8d%2A%22%29&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=desc&viz=stream&from_ts=1770839417617&to_ts=1770839481773&live=false)
    - SMS Delivered `host:*dev* ("*ada68369-851d-4d87-a93e-72879b579bc7*" OR @requestId:ada68369-851d-4d87-a93e-72879b579bc7 OR "*bc9e95c2-0130-452e-a8f8-7c75fb0473c5*") `
    - EMAIL Delivered `host:*dev* ("*18f9498a-56b6-440c-80b5-73caf34bf9ad*" OR @requestId:18f9498a-56b6-440c-80b5-73caf34bf9ad) `

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

**Comment by ChrisJohnson-CDJ** [comment]
- Retry logic and tests are in place
- Reviewing unit test coverage and checking for degenerate edge cases
- Posted request for review of new RetryableException_NonPriority
- DEV deploy and final testing tomorrow (2/11)

File: app/celery/provider_tasks.py [hunk 1]
```diff
@@ -301,12 +305,28 @@ def _handle_delivery_failure(  # noqa: C901 - too complex (11 > 10)
...
-            raise AutoRetryException(f'Found {type(e).__name__}, autoretrying...', e, e.args)
+            if isinstance(e, RetryableException) and e.use_non_priority_handling:
+                # Handle non-priority RetryableException by retrying in 'retry-tasks' queue
+                # This queue is serviced by the non-priority worker pool and lessens impact on priority tasks
+
+                current_app.logger.warning(
+                    '%s unable to send for notification %s, retrying in non-priority worker pool',
+                    notification_type,
+                    notification_id,
+                )
```
[the remark this answers is not in the document]
Comment (ChrisJohnson-CDJ) [reply app/celery/provider_tasks.py]: Added queue name to log message
